app/src/main/kotlin/com/ark/globe/repositories/Repository.kt

```diff
@@ -0,0 +1,182 @@
+            else null
+        }
+        else null
+    }
+
+    private fun loadUrl(url: String?){
+        val client: OkHttpClient
+        val request: Request
+        var response: okhttp3.Response? = null
+        if(url != null) {
+            client = OkHttpClient.Builder().build()
+            request = Request.Builder()
+                .url(url)
+                .build()
+            try {
+                response = client.newCall(request).execute()
+
+                val bReader = BufferedReader(response.body?.charStream())
+                val status = response.code
+                var actualUrl = ""
+                Log.d("Status", status.toString())
+                var isParsed = false
+                if (status == HttpURLConnection.HTTP_OK) {
+                    actualUrl = response.request.url.toString()
+                    Log.d("Actual URL", actualUrl)
+                    bReader.forEachLine {
+                        //println("Line: $it")
+                        if (it.contains("/@"))
+                            Log.d("Redirect", it.substringAfter("/@"))
+                        isParsed = parseGoogleCoordinates(it)
+                        if (isParsed) {
+                            Log.d("Is Parsed", isParsed.toString())
+                            return@forEachLine
+                        }
+                    }
+                }
+//  https://maps.app.goo.gl/XTudbauXz5ZdfXAWA
+//  https://www.google.com/maps?q=Google+Building+1600,+1600+Plymouth+St,+Mountain+View,+CA+94043,+USA&ftid=0x808fba002c047109:0x8a6e9df8c478269&hl=en&gl=us&g_ep=GAA%3D&shorturl=1https://www.google.com/maps?q=Google+Building+1600,+1600+Plymouth+St,+Mountain+View,+CA+94043,+USA&ftid=0x808fba002c047109:0x8a6e9df8c478269&hl=en&gl=us&g_ep=GAA%3D&shorturl=1https://www.google.com/maps?q=Google+Building+1600,+1600+Plymouth+St,+Mountain+View,+CA+94043,+USA&ftid=0x808fba002c047109:0x8a6e9df8c478269&hl=en&gl=us&g_ep=GAA%3D&shorturl=1
+                if(!isParsed) {
+                    Log.d("Is parsed", isParsed.toString())
+                    if (actualUrl.contains(GOOGLE_MAPS)) {
+                        parseGoogleMapsLinks(actualUrl)
+                        return
+                    }
+
+                    if (actualUrl.contains(OPEN_STREET_MAP)) {
+                        parseOpenStreetMapLinks(actualUrl)
+                        return
+                    }
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+            } finally {
+                response?.close()
+            }
+        }
+    }
+
+    private fun parseGoogleCoordinates(line: String): Boolean{
+        return if(line.contains(DELIMITER_1)){
+            coordinatesString = line.substringAfter(DELIMITER_1).substringBefore(DELIMITER_3)
+            val coords = coordinatesString.split(DELIMITER_2)
+            latitude = coords[0]
+            longitude = coords[1]
+            println("Start: $coordinatesString")
```

Comment:
Method `println` should not be used for anything except debugging.

Reply:
Alright